### ISSUE TITLE
Report full amount of tests

### ISSUE THREAD

**Issue by Gormartsen** [backdrop/backdrop-issues#1963]
Guys!

We have **50590** tests!!!
That's unbelievable amount to me.

I sent [PR 1441](https://github.com/backdrop/backdrop/pull/1441) to add total_count to summary(travis) status.
<img width="701" alt="screen shot 2016-06-10 at 11 30 08 am" src="https://cloud.githubusercontent.com/assets/5018896/15969598/bcdc4476-2efe-11e6-80e5-a21b6a5946be.png">
<img width="702" alt="screen shot 2016-06-10 at 11 05 34 am" src="https://cloud.githubusercontent.com/assets/5018896/15968803/4c79fa82-2efb-11e6-97ac-25df1d486341.png">

But now I am curios why it's different number for total. Anyone?

**Comment by Gormartsen** [comment]
Is it necessary to have a dot at the end of the message? Do we need it ?

**Comment by docwilmot** [comment]
> We have 50590 tests!!!

I don't think so. Search for string `function testXXX` in core gets only 1360 hits.

**Comment by Gormartsen** [comment]
@docwilmot you are right. This code does not work:
```
 $total_count = db_query("SELECT COUNT(*) FROM {simpletest} WHERE test_id = :test_id ", array(':test_id' => $test_id))->fetchField();
```

**Comment by Gormartsen** [comment]
Got it. It's debug messages mostly

**Comment by Gormartsen** [comment]
@docwilmot I updated PR. So we have around 40k `$this->assert*`
That's what it count. Each test does many asserts. It's basically not a count of test methods but tests that we are doing (asserts)

**Comment by docwilmot** [comment]
Understood.

Is that going to be useful information though? If the test crashes, it wouldnt finish the asserts in that test, so the total number of asserts would vary. I think which `testXXX` failed is more useful if youre not already doing that.

Unless youre planning to also say which asserts failed (which may be helpful sometimes for debugging)

**Comment by Gormartsen** [comment]
> Is that going to be useful information though?

That's a question to all of us. It has no more meaning than count of asserts we check.
I am playing around to make report page more useful for us. Any propositions are welcome as well.
You can [post them as issues here](https://github.com/Zen-CI/issues/issues).

@docwilmot tests are going forward after asserts failed. At least usually:
```
        $this->assertNoFieldByName('op', t('Preview'), 'Preview button not found.');
        $this->assertFieldByName('op', t('Save'), 'Save button found.');
        $this->backdropPost(NULL, $edit, t('Save'));
```

But it is true that this value is not constant.

> Unless youre planning to also say which asserts failed 

We already has this info somehow represented in report header :
https://zen.ci/backdrop/backdrop/test/test-php53-backdrop_backdrop_1441-7208

```
Field API: Field translations tests (FieldTranslationsTestCase)

fail: The test did not complete due to a fatal error. field.test:3023
```

**Comment by docwilmot** [comment]
> @docwilmot tests are going forward after asserts failed. At least usually:

I meant crash, like fatal error in that test. 

> We already has this info somehow represented in report header :

I see where it says which tests failed but not the individual assert(s) that caused the fail. When you run tests through the UI it gives a line by line account of every assert, every debug for each test. I'm not saying we need that, 
![test](https://cloud.githubusercontent.com/assets/1042283/15972535/05d0ff1a-2f0d-11e6-8655-df7d9a6bb402.PNG)
but this is what I meant.

But regardless, I am not claiming that the count has no value, just whether its worth your time to do it. And I agree it may be useful to others.

**Comment by Gormartsen** [comment]
Tnx @docwilmot I was just it happy with 535 number:) it's a classes amount:)
